hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/api/StorageDistributionEndpoint.java

```diff
@@ -0,0 +1,321 @@
+    result.put("pendingKeySize", pendingKeySize.getOrDefault("totalReplicatedDataSize", 0L));
+    return result;
+  }
+
+  private long calculateOpenKeySizes() {
+    Map<String, Long> openKeySummary = new HashMap<>();
+    omdbInsightEndpoint.createKeysSummaryForOpenKey(openKeySummary);
+    Map<String, Long> openKeyMPUSummary = new HashMap<>();
+    omdbInsightEndpoint.createKeysSummaryForOpenMPUKey(openKeyMPUSummary);
+    long openKeyDataSize = openKeySummary.getOrDefault("totalReplicatedDataSize", 0L);
+    long totalMPUKeySize = openKeyMPUSummary.getOrDefault("totalReplicatedDataSize", 0L);
+    return openKeyDataSize + totalMPUKeySize;
+  }
+
+  private long calculateCommittedSize() {
+    try {
+      Response rootResponse = nsSummaryEndpoint.getDiskUsage("/", false, true, false);
+      if (rootResponse.getStatus() != Response.Status.OK.getStatusCode()) {
+        log.warn("Failed to get disk usage, status: {}", rootResponse.getStatus());
+        return 0L;
+      }
+      DUResponse duRootRes = (DUResponse) rootResponse.getEntity();
+      return duRootRes != null ? duRootRes.getSizeWithReplica() : 0L;
+    } catch (IOException e) {
+      log.error("IOException while calculating committed size", e);
+      return 0L;
+    }
+  }
+
+  private DeletionPendingBytesByStage createDeletionPendingBytesByStage(long pendingDirectorySize,
+                                                                        long pendingKeySize,
+                                                                        long totalPendingAtScmSide,
+                                                                        long totalPendingAtDnSide) {
+    long totalPending = pendingDirectorySize + pendingKeySize + totalPendingAtScmSide + totalPendingAtDnSide;
+    Map<String, Map<String, Long>> stageItems = new HashMap<>();
+    Map<String, Long> omMap = new HashMap<>();
+    omMap.put("totalBytes", pendingDirectorySize + pendingKeySize);
+    omMap.put("pendingDirectoryBytes", pendingDirectorySize);
+    omMap.put("pendingKeyBytes", pendingKeySize);
+    Map<String, Long> scmMap = new HashMap<>();
+    scmMap.put("pendingBytes", totalPendingAtScmSide);
+    Map<String, Long> dnMap = new HashMap<>();
+    dnMap.put("pendingBytes", totalPendingAtDnSide);
+    stageItems.put("OM", omMap);
+    stageItems.put("SCM", scmMap);
+    stageItems.put("DN", dnMap);
+    return new DeletionPendingBytesByStage(totalPending, stageItems);
+  }
+
+  private void initializeBlockDeletionMetricsMap() {
+    nodeManager.getNodeStats().keySet().forEach(nodeId -> {
+      try {
+        long dnPending = ReconUtils.getMetricsFromDatanode(nodeId,
+                "HddsDatanode",
+                "BlockDeletingService",
+                "TotalPendingBlockBytes");
+        blockDeletionMetricsMap.put(nodeId, dnPending);
+      } catch (IOException e) {
+        throw new RuntimeException(e);
```

Comment:
We need not to catch the exception , in fact we can remove try catch as you are already handling it in caller. Here we are simply catching and then throwing back. It is redundant.